### PR TITLE
rest-api: a number of improvements to the rest api

### PR DESCRIPTION
**Pull request by bmwill**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-rest-api/src/info.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct NodeInfo {
+    pub chain_id: ChainIdentifier,
+    pub epoch: u64,
+    pub checkpoint_height: u64,
+    pub timestampe_ms: u64,
+    pub oldest_checkpoint_height: u64,
+    pub node_type: NodeType,
+    pub git_revision: Cow<'static, str>,
```
Comment (sadhansood) [inline crates/sui-rest-api/src/info.rs:44]: Does it also make sense to return protocol version node is running with?

File: crates/sui-rest-api/src/info.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+    .pipe(Ok)
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct NodeInfo {
+    pub chain_id: ChainIdentifier,
+    pub epoch: u64,
+    pub checkpoint_height: u64,
+    pub timestampe_ms: u64,
```
Comment (sadhansood) [inline crates/sui-rest-api/src/info.rs:41]: type: s/timestampe/timestamp

**Review by sadhansood — APPROVED** [review]
LGTM

File: crates/sui-rest-api/src/info.rs (hunk 1 above)
Comment (bmwill) [reply crates/sui-rest-api/src/info.rs:44]: yeah that's a great idea!

**Pull request merged**